app/api/src/docker/openscad/openscad.js

```diff
@@ -50,39 +77,46 @@ const preview = async (req, _context, callback) => {
   })
 }
 
-// const stl = async (req, _context, callback) => {
-//   _context.callbackWaitsForEmptyEventLoop = false
-//   const eventBody = Buffer.from(req.body, 'base64').toString('ascii')
-//   console.log(eventBody, 'eventBody')
-//   const { file } = JSON.parse(eventBody)
-//   const { error, result, tempFile } = await stlExport({ file })
-//   if (error) {
-//     const response = {
-//       statusCode: 400,
-//       body: { error, tempFile },
-//     }
-//     callback(null, response)
-//   } else {
-//     console.log(`got result in route: ${result}, file is: ${tempFile}`)
-//     const fs = require('fs')
-//     const stl = fs.readFileSync(`/tmp/${tempFile}/output.stl`, {
-//       encoding: 'base64',
-//     })
-//     console.log('encoded stl', stl)
-//     const response = {
-//       statusCode: 200,
-//       headers: {
-//         'content-type': 'application/stl',
-//       },
-//       body: stl,
-//       isBase64Encoded: true,
-//     }
-//     console.log('callback fired')
-//     callback(null, response)
-//   }
-// }
+const stl = async (req, _context, callback) => {
+  _context.callbackWaitsForEmptyEventLoop = false
+  const eventBody = req.body
+  console.log(eventBody, 'eventBody')
+  const stlKey = openScadStlKey(eventBody)
+
+  console.log('key', stlKey)
+
+  const params = {
+    Bucket: process.env.BUCKET,
+    Key: stlKey,
+  }
+  console.log('original params', params)
+  const previousAsset = await checkIfAlreadyExists(params, s3)
+  if (previousAsset.isAlreadyInBucket) {
+    console.log('already in bucket')
+    const response = {
+      statusCode: 200,
+      body: JSON.stringify({
+        url: getObjectUrl({ ...params }, s3),
+        consoleMessage: previousAsset.consoleMessage,
+      }),
+    }
+    callback(null, response)
+    return
+  }
+  const { file } = JSON.parse(eventBody)
+  const { error, consoleMessage, fullPath } = await stlExport({ file })
```

Comment:
I'm not getting console messages from the stl generation, I've had a go a debugging it but it seems like openscad just isn't outputting anything. Might have to dive deeper again later.